extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/ThrowingSupplier.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.tracer;
+
+import java.util.function.Supplier;
+
+/**
+ * A utility interface representing a {@link Supplier} that may throw.
+ *
+ * @param <E> Thrown exception type.
+ */
+@FunctionalInterface
+public interface ThrowingSupplier<T, E extends Throwable> {
+  T get() throws E;
```

Comment:
since this is our interface, we could potentially give it more descriptive class/method names

a possible option, borrowing from Spring's [TransactionCallback/doInTransaction](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/reactive/TransactionCallback.html):

```suggestion
public interface SpanCallback<T, E extends Throwable> {
  T doInSpan() throws E;
```

Reply:
great idea :smile: